org.hl7.fhir.r5/src/main/java/org/hl7/fhir/r5/terminologies/utilities/ValidationResult.java

```diff
@@ -67,7 +67,23 @@ public ValidationResult(IssueSeverity severity, String message, List<OperationOu
     if (issues != null) {
       this.issues.addAll(issues);
       for (OperationOutcomeIssueComponent issue : issues) {
-        if (issue.getSeverity() == OperationOutcome.IssueSeverity.ERROR || message == null) {
+        if (issue.getSeverity() == OperationOutcome.IssueSeverity.ERROR) {
+          String msg = issue.getDetails().getText();
+          if (!this.messages.contains(msg)) {
+            this.messages.add(msg);
+          }
+        }
+      }
+      for (OperationOutcomeIssueComponent issue : issues) {
+        if (issue.getSeverity() == OperationOutcome.IssueSeverity.WARNING && messages.isEmpty()) {
```

Comment:
Why the `isEmpty()` check here and below? 

I assume this is so that if there are errors, there are only errors in messages, and if no errors, then only warnings, and finally only adding info if there's nothing else of greater severity.

For testing purposes, I assume we should only be investigating issue lists then, and treating messages as a courtesy?

Reply:
no we care about both - the message is a summary of the errors if there are errors. Else warnings. Else hints. This code was disagreeing with the rest of the code, and leading to irregular message outcomes. fixed now